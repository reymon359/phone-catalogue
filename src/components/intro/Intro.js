import React, { Component } from 'react';


class Intro extends Component {
    shouldComponentUpdate(nextProps, nextState) {
        return false;
    }
    render() {
        return (
            <div>
               <p className="f4 near-white pb4">
                    Project done by  <a className="link underline near-white hover-navy" href="https://ramonmorcillo.com/">Ram√≥n Morcillo</a> 
                    for the <a  className="link underline near-white hover-navy" href="https://www.guidesmiths.com/">Guidesmiths</a> interview code 
                    challenge about React ( <a  className="link underline near-white hover-navy" href="https://github.com/guidesmiths/interview-code-challenges/blob/master/react/phone-catalogue">more info here</a> ). 
                    You can find the source code <a  className="link underline near-white hover-navy" href="https://github.com/reymon359/phone-catalogue">here</a>.
                </p>
            </div>
        );
    }
}

export default Intro;