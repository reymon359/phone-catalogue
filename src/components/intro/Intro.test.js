import { shallow } from 'enzyme';
import React from 'react';
import Intro from './Intro';

it('should render Intro component', () => {
    expect(shallow(<Intro />)).toMatchSnapshot();
});
