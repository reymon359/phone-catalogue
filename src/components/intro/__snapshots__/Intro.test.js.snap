// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Intro component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Intro />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="f4 near-white pb4"
      >
        Project done by  
        <a
          className="link underline near-white hover-navy"
          href="https://ramonmorcillo.com/"
        >
          Ramón Morcillo
        </a>
        for the 
        <a
          className="link underline near-white hover-navy"
          href="https://www.guidesmiths.com/"
        >
          Guidesmiths
        </a>
         interview code challenge about React ( 
        <a
          className="link underline near-white hover-navy"
          href="https://github.com/guidesmiths/interview-code-challenges/blob/master/react/phone-catalogue"
        >
          more info here
        </a>
         ). You can find the source code 
        <a
          className="link underline near-white hover-navy"
          href="https://github.com/reymon359/phone-catalogue"
        >
          here
        </a>
        .
      </p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Project done by  ",
          <a
            className="link underline near-white hover-navy"
            href="https://ramonmorcillo.com/"
          >
            Ramón Morcillo
          </a>,
          "for the ",
          <a
            className="link underline near-white hover-navy"
            href="https://www.guidesmiths.com/"
          >
            Guidesmiths
          </a>,
          " interview code challenge about React ( ",
          <a
            className="link underline near-white hover-navy"
            href="https://github.com/guidesmiths/interview-code-challenges/blob/master/react/phone-catalogue"
          >
            more info here
          </a>,
          " ). You can find the source code ",
          <a
            className="link underline near-white hover-navy"
            href="https://github.com/reymon359/phone-catalogue"
          >
            here
          </a>,
          ".",
        ],
        "className": "f4 near-white pb4",
      },
      "ref": null,
      "rendered": Array [
        "Project done by  ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Ramón Morcillo",
            "className": "link underline near-white hover-navy",
            "href": "https://ramonmorcillo.com/",
          },
          "ref": null,
          "rendered": "Ramón Morcillo",
          "type": "a",
        },
        "for the ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Guidesmiths",
            "className": "link underline near-white hover-navy",
            "href": "https://www.guidesmiths.com/",
          },
          "ref": null,
          "rendered": "Guidesmiths",
          "type": "a",
        },
        " interview code challenge about React ( ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "more info here",
            "className": "link underline near-white hover-navy",
            "href": "https://github.com/guidesmiths/interview-code-challenges/blob/master/react/phone-catalogue",
          },
          "ref": null,
          "rendered": "more info here",
          "type": "a",
        },
        " ). You can find the source code ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "here",
            "className": "link underline near-white hover-navy",
            "href": "https://github.com/reymon359/phone-catalogue",
          },
          "ref": null,
          "rendered": "here",
          "type": "a",
        },
        ".",
      ],
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="f4 near-white pb4"
        >
          Project done by  
          <a
            className="link underline near-white hover-navy"
            href="https://ramonmorcillo.com/"
          >
            Ramón Morcillo
          </a>
          for the 
          <a
            className="link underline near-white hover-navy"
            href="https://www.guidesmiths.com/"
          >
            Guidesmiths
          </a>
           interview code challenge about React ( 
          <a
            className="link underline near-white hover-navy"
            href="https://github.com/guidesmiths/interview-code-challenges/blob/master/react/phone-catalogue"
          >
            more info here
          </a>
           ). You can find the source code 
          <a
            className="link underline near-white hover-navy"
            href="https://github.com/reymon359/phone-catalogue"
          >
            here
          </a>
          .
        </p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Project done by  ",
            <a
              className="link underline near-white hover-navy"
              href="https://ramonmorcillo.com/"
            >
              Ramón Morcillo
            </a>,
            "for the ",
            <a
              className="link underline near-white hover-navy"
              href="https://www.guidesmiths.com/"
            >
              Guidesmiths
            </a>,
            " interview code challenge about React ( ",
            <a
              className="link underline near-white hover-navy"
              href="https://github.com/guidesmiths/interview-code-challenges/blob/master/react/phone-catalogue"
            >
              more info here
            </a>,
            " ). You can find the source code ",
            <a
              className="link underline near-white hover-navy"
              href="https://github.com/reymon359/phone-catalogue"
            >
              here
            </a>,
            ".",
          ],
          "className": "f4 near-white pb4",
        },
        "ref": null,
        "rendered": Array [
          "Project done by  ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ramón Morcillo",
              "className": "link underline near-white hover-navy",
              "href": "https://ramonmorcillo.com/",
            },
            "ref": null,
            "rendered": "Ramón Morcillo",
            "type": "a",
          },
          "for the ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Guidesmiths",
              "className": "link underline near-white hover-navy",
              "href": "https://www.guidesmiths.com/",
            },
            "ref": null,
            "rendered": "Guidesmiths",
            "type": "a",
          },
          " interview code challenge about React ( ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "more info here",
              "className": "link underline near-white hover-navy",
              "href": "https://github.com/guidesmiths/interview-code-challenges/blob/master/react/phone-catalogue",
            },
            "ref": null,
            "rendered": "more info here",
            "type": "a",
          },
          " ). You can find the source code ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "here",
              "className": "link underline near-white hover-navy",
              "href": "https://github.com/reymon359/phone-catalogue",
            },
            "ref": null,
            "rendered": "here",
            "type": "a",
          },
          ".",
        ],
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
